### PR TITLE
[ci] Make rspec webui tests work

### PR DESCRIPTION
**Pull request by shyukri**
- Need to put a lot of safety checks
- wait for ajax helper was useless because `jQuery.active` check was not reliable when pulling the live log.
- also fixed Capybara::DSL warning.

Working example:
https://openqa.opensuse.org/tests/overview?distri=obs&version=unstable&build=2.6.51-24.38&groupid=17

### CODE REVIEW

**Comment by shyukri** [comment]
@ChrisBr , @hennevogel pls review

File: dist/t/spec/features/login_spec.rb [hunk 1]
```diff
@@ -75,19 +75,46 @@
...
+      Timeout.timeout(160) {
+        if page.has_content?("No live log available:") then
+          page.evaluate_script("window.location.reload()")
+          first(:link, "Start refresh").click
+        end
+        expect(page).to have_selector("div#log_space_wrapper", :wait => 20)
+        expect(page).to have_content('finished "build build.spec"', :wait => 160)
+      }
+    rescue Timeout::Error
+      puts "Let's retry on rescue."
```
Comment (ChrisBr) [inline dist/t/spec/features/login_spec.rb]: What is with the puts? Can we somehow write it to a log instead?

File: dist/t/spec/features/login_spec.rb [hunk 2]
```diff
@@ -75,19 +75,46 @@
...
     visit "/package/live_build_log/home:Admin/obs-build/openSUSE_Tumbleweed/i586"
-    wait_for_ajax
+    begin
+      Timeout.timeout(160) {
+        if page.has_content?("No live log available:") then
+          page.evaluate_script("window.location.reload()")
+          first(:link, "Start refresh").click
+        end
+        expect(page).to have_selector("div#log_space_wrapper", :wait => 20)
+        expect(page).to have_content('finished "build build.spec"', :wait => 160)
```
Comment (ChrisBr) [inline dist/t/spec/features/login_spec.rb]: You need to specify a wait if it's already wrapped in a timeout(160) block?

File: dist/t/spec/features/login_spec.rb [hunk 3]
```diff
@@ -75,19 +75,46 @@
...
-    wait_for_ajax
+    begin
+      Timeout.timeout(90) {
+        expect(page).to have_selector("div#log_space_wrapper", :wait => 20)
+        page.evaluate_script("window.location.reload()") if page.has_content?("No live log available:")
+        next if page.has_content?("Build finished")
+        expect(page).to have_content('finished "build build.spec"', :wait => 90)
+      }
+    rescue Timeout::Error
+      puts "Let's retry on rescue."
```
Comment (ChrisBr) [inline dist/t/spec/features/login_spec.rb]: Same here!

File: dist/t/spec/features/login_spec.rb (hunk 1 above)
Comment (shyukri) [inline dist/t/spec/features/login_spec.rb]: if the job fails in openQA, i'm uploading /srv/[obs|www] , but yeah we can put it on file

File: dist/t/spec/spec_helper.rb [hunk 4]
```diff
@@ -19,6 +18,7 @@
   end
   # Limits the available syntax to the non-monkey patched
   config.disable_monkey_patching!
+  config.include Capybara::DSL
```
Comment (ChrisBr) [inline dist/t/spec/spec_helper.rb]: Do we really need to include it for all tests?

Comment (shyukri) [inline dist/t/spec/spec_helper.rb]: It should apply magically only on features: http://stackoverflow.com/questions/18616683/why-do-i-get-including-capybaradsl-in-the-global-scope-is-not-recommended?answertab=votes#tab-top

File: dist/t/spec/features/login_spec.rb (hunk 1 above)
Comment (ChrisBr) [inline dist/t/spec/features/login_spec.rb]: What has uploading to do with the puts?

File: dist/t/spec/spec_helper.rb (hunk 4 above)
Comment (ChrisBr) [inline dist/t/spec/spec_helper.rb]: But then you should c&p right ;)
``
config.include Capybara::DSL, :type => :feature
``

Comment (ChrisBr) [inline dist/t/spec/spec_helper.rb]: Hmm seems the type option is no longer valid ...

File: dist/t/spec/features/login_spec.rb (hunk 2 above)
Comment (shyukri) [inline dist/t/spec/features/login_spec.rb]: I'm rescuing the Timeout failure only, that's why wait has the same amount

File: dist/t/spec/spec_helper.rb (hunk 4 above)
Comment (hennevogel) [inline dist/t/spec/spec_helper.rb]: The question is rather what is the difference to how it was before?

**Comment by shyukri** [comment]
I've removed the puts, can you review again?

**Comment by hennevogel** [comment]
@shyukri can you answer the other questions? :-)

File: dist/t/spec/spec_helper.rb (hunk 4 above)
Comment (shyukri) [inline dist/t/spec/spec_helper.rb]:
```
require 'capybara'
......
include Capybara::DSL
```
resulting in `including Capybara::DSL in the global scope is not recommended!`

Comment (hennevogel) [inline dist/t/spec/spec_helper.rb]: I see

**Comment by hennevogel** [comment]
Then let's go :-)

**Pull request merged**